From 8d34bbaab8da2cbbd7d5479e2980301cf580e84d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Thu, 24 Oct 2024 17:42:56 +0200
Subject: [PATCH 736/954] feat(stm32mp2): force boot interface instance on
 M33TD eMMC
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Boot interface instance must be forced to 2 in case of eMMC
single boot device to avoid a ROM issue when cortex M33 is TDCID.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git bb1412f4deec6ff85c762b71faf9030a2183f616 feat(stm32mp2): force boot interface instance on M33TD eMMC]
Change-Id: I59b91e35cc5cf9ea39ff59b392c62a73936d7bb0
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/413980
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/bl2_plat_setup.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index b0b55edcb0..dcfbc2ce5e 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -417,6 +417,16 @@ void bl2_el3_plat_arch_setup(void)
 
 	stm32_iwdg_refresh();
 
+#if STM32MP_M33_TDCID
+	/*
+	 * boot interface instance must be forced to 2 in case of eMMC
+	 * single boot device to avoid a ROM issue when M33 is TDCID.
+	 */
+	if (boot_context->boot_interface_selected == BOOT_API_CTX_BOOT_INTERFACE_SEL_FLASH_EMMC) {
+		boot_context->boot_interface_instance = 2U;
+	}
+#endif
+
 	stm32_save_boot_info(boot_context);
 
 	/* Masking potential tamper during BL2 */
-- 
2.53.0

